ocrd/ocrd/workspace.py

```diff
@@ -688,6 +695,10 @@ def image_from_page(self, page, page_id,
                     fill=fill, transparency=transparency)
 
         # verify constraints again:
+        if filename and not getattr(page_image, 'filename', '').endswith(filename):
```

Comment:
page_image.filename is always empty for me, so this check always fails

Reply:
> page_image.filename is always empty for me, so this check always fails

then that AlternativeImage is not **maximal**, i.e. there are operations (derotation, cropping) that are possible from the description of the segment but have not been applied to the image yet – so the API does it automatically before returning an image. Any PIL.Image operation will invalidate the `filename` attribute, though.

Can you try retrieving the image with additional filters like `feature_filter=deskewed`?

Could you share the snipped of the segment/page that shows the AlternativeImage elements with their comments?

Reply:
I agree it's somewhat unexpected, but we have no other choice from the API point of view. Other callers would be surprised to _not_ get an image that corresponds with the features of the segment.

But we could clarify that in the docstring...